---
title: Resources for Working with Single-Cell RNA-seq
---

On this page, we've assembled some resources you may find helpful for working with single-cell RNA-seq data.

**Table of contents**

* TOC goes here
{:toc}

## Obtaining practice datasets


### The Single-Cell Pediatric Cancer Atlas (ScPCA)

The Childhood Cancer Data Lab builds and maintains the [Single-cell Pediatric Cancer Atlas (ScPCA) Portal](https://scpca.alexslemonade.org), a resource of single-cell/nuclei transcriptomic data generated by ALSF-funded labs and uniformly processed by the Data Lab.
The data here has been quantified and processed into Bioconductor `SingleCellExperiment` objects, saved as `.rds` files, and are ready for downstream analysis using the tools we have shown in this workshop.
You can read more about how we process data in ScPCA and how you can use ScPCA datasets in [our documentation](https://scpca.readthedocs.io/en/latest/).


#### The `scRNAseq` Bioconductor package

The [`scRNAseq`](https://bioconductor.org/packages/release/data/experiment/html/scRNAseq.html) Bioconductor package contains [dozens of scRNA-seq datasets](https://bioconductor.org/packages/release/data/experiment/vignettes/scRNAseq/inst/doc/scRNAseq.html) formatted as `SingleCellExperiment` objects.


### _Tabula Muris_ data

This is a more extensive set of the [_Tabula Muris_](https://tabula-muris.ds.czbiohub.org) data (mouse tissue) that are used in our "Introduction to scRNA-seq" training.
These samples, already processed by `salmon alevin`, can be found in the `~/shared-data/training-data/tabula-muris/alevin` directory.
Metadata, including tissue of origin for each sample (since the sample names themselves are not informative), can be found in `~/training-modules/scRNA-seq/data/tabula-muris/TM_droplet_metadata.csv`.
Note that this data is given at the cell level: simplifying the table to the sample level is a good opportunity to practice some data wrangling skills!
(It is also a `CSV` file; don't forget to use `readr::read_csv()` when loading it!)


### Human Cell Atlas data

Another potential source for processed single cell data is the [Human Cell Atlas (HCA) Data Portal](https://data.humancellatlas.org/).
The data here is from a mix of technologies, including both 10X, Smart-seq2, and DropSeq.
The HCA has standardized processing pipelines for 10X and Smart-seq2, though it seems that most of the processed data is 10X, so we recommend focusing on those projects.

To download a data set, first browse or search to find a project of interest.
Click on the project name to see an abstract and other information for the project.

You can then select "Project Matrices" from the left side to download the processed single-cell expression data.
Scroll down to the "DCP Generated Matrices" section on the "Project Matrices" page, as the data here will be uniformly processed and in a standard data format.
That format is called `loom`, and we can read it into `R` in a fairly straightforward way.
Once you find a loom file listed (not all projects have one, unfortunately), you have two options:

1. Click the "Copy download link" button (the tiny clipboard icon) and then use that URL to download the file directly to the RStudio server [following these instructions](../workshop/working-with-your-own-data.md#loading-data-from-a-website).
Be sure to put quotes around the _very long_ URL that is provided, and specify a filename for the download with the `-O` option.

1. Download the loom file to your computer (look for the tiny icon with the arrow pointing down) and upload it to the server [following these instructions](../workshop/working-with-your-own-data.md#uploading-small-files-to-rstudio-server).

#### Reading `loom` format data in R

Once you have a `.loom` file on the server, you can use the following commands in R to import the data as a `SingleCellExperiment`-compatible object.

```r
loomfile <-  file.path("path", "to", "file.loom")
sce <- LoomExperiment::import(loomfile, type = "SingleCellLoomExperiment")
# the first assay matrix should be named "counts"
assayNames(sce)[1] <- "counts"
```

The last command is to be sure that the main data matrix, which contains count data, has the name that the `SingleCellExperiment` commands expect.

The gene and cell identifiers are stored in `rowData` and `colData` respectively, but those identifiers aren't used as row names and column names.
To make the format a little closer to what we work with during instruction (and so we can visualize individual genes), we need to do the following:

```r
rownames(sce) <- rowData(sce)$Gene
colnames(sce) <- colData(sce)$CellID
```

Once that is done, all of the `SingleCellExperiment` commands that we have demonstrated should work!
You will want to be sure to look at `rowData()` and `colData()`, as some of the contents will be different from what we have seen in previous data sets (and may vary among projects).
Some of the QC calculations may have already been performed, but the data will not be filtered or normalized.
You will need to perform those steps on your own.

## Additional single-cell RNA-seq resources

This list provides some links to external resources on single-cell RNA-seq analysis methods that may be useful to you as you develop your own single-cell RNA-seq analysis skills and practices.
Please note, this is not an exhaustive list!
It includes multiple types of resources for various topics in single-cell RNA-seq analysis, but does not represent the complete breadth of analysis topics.
Resources are listed by topic and in alphabetical order, not in order of recommendation.

### General Single-cell resources

- [An introduction to the SingleCellExperiment class - Bioconductor](https://www.bioconductor.org/packages/devel/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html)
- [Analysis of single cell RNA-seq data - Hemburg Lab](https://www.singlecellcourse.org/)
- [Best practices for single-cell analysis across modalities - Heumos *et al.* (2023)](https://doi.org/10.1038/s41576-023-00586-w)
- [Orchestrating Single-cell Analysis (OSCA) with Bioconductor - Bioconductor](https://bioconductor.org/books/release/OSCA/)

### Alignment and quantification of gene expression

- [A like-for-like comparison of lightweight-mapping pipelines for single-cell RNA-seq data pre-processing - Zakeri _et al._ (2021)](https://doi.org/10.1101/2021.02.10.430656)
- [Alevin-fry unlocks rapid, accurate and memory-frugal quantification of single-cell RNA-seq data - He _et al._ (2022)](https://doi.org/10.1038/s41592-022-01408-3)
- [Cell Ranger Overview - 10X Genomics](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger)
- [Comparative analysis of common alignment tools for single-cell RNA sequencing - Bruning _et al._ (2022)](https://doi.org/10.1093/gigascience/giac001)

### Filtering and normalization

- [EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data - Lun _et al._ (2019)](https://doi.org/10.1186/s13059-019-1662-y)
- [miQC: An adaptive probabilistic framework for quality control of single-cell RNA-sequencing data - Hippen _et al._ (2021)](https://doi.org/10.1371/journal.pcbi.1009290)
- [OSCA Basics](http://bioconductor.org/books/release/OSCA.basic/)
- [Utilities for handling droplet-based single-cell RNA-seq data: Detecting empty droplets](https://bioconductor.org/packages/devel/bioc/vignettes/DropletUtils/inst/doc/DropletUtils.html#detecting-empty-droplets)

### Dimensionality reduction and clustering

- [OSCA chapter on clustering](http://bioconductor.org/books/release/OSCA.basic/clustering.html)
- [OSCA chapter on clustering metrics](http://bioconductor.org/books/release/OSCA.advanced/clustering-redux.html)
- [OSCA chapter on dimensionality reduction](http://bioconductor.org/books/release/OSCA.basic/dimensionality-reduction.html)
- [PCA - Principal Component Analysis](http://www.nlpca.org/pca_principal_component_analysis.html)
- [Principal Component Analysis - A Brief Introduction](https://medium.com/x8-the-ai-community/principal-component-analysis-a-brief-introduction-dc8cf3e03c71)

### Cell type annotation

- [AUCell: Identifying cells with active gene sets](https://bioconductor.org/packages/release/bioc/vignettes/AUCell/inst/doc/AUCell.html)
- [Azimuth](https://azimuth.hubmapconsortium.org/)
- [Identifying cell types to interpret scRNA-seq data: how, why and more possibilities - Wang _et al._ (2020)](https://doi.org/10.1093/bfgp/elaa003)
- [OSCA chapter on cell type annotation](https://bioconductor.org/books/release/OSCA.basic/cell-type-annotation.html)
- [scType](http://sctype.app)
- [The SingleR Book - Bioconductor](https://bioconductor.org/books/release/SingleRBook/)
- [Web resources for cell type annotation - 10X Genomics](https://www.10xgenomics.com/resources/analysis-guides/web-resources-for-cell-type-annotation)

### CITE-seq

- [OSCA chapter on integrating with protein abundance](http://bioconductor.org/books/release/OSCA.advanced/integrating-with-protein-abundance.html)
- [Simultaneous epitope and transcriptome measurement in single cells - Stoeckius _et al._ (2017)](https://doi.org/10.1038/nmeth.4380)

### Integrating scRNA-seq samples

- [A description of the theory behind the `fastMNN` algorithm](https://marionilab.github.io/FurtherMNN2018/theory/description.html)
- [Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors - Haghverdi _et al._ (2018)](https://doi.org/10.1038/nbt.4091)
- [Benchmarking atlas-level data integration in single-cell genomics - Luecken _et al._ (2021)](https://doi.org/10.1038/s41592-021-01336-8)
- [Fast, sensitive and accurate integration of single-cell data with Harmony - Korsunsky _et al._ (2019)](https://doi.org/10.1038/s41592-019-0619-0)
- [OSCA section on multi-sample analyses](http://bioconductor.org/books/release/OSCA.multisample/)

### Differential expression analysis

- [Batch effects and the effective design of single-cell gene expression studies - Tung _et al._ (2017)](https://doi.org/10.1038/srep39921)
- [Differential gene expression analyses in scRNA-seq data between conditions with biological replicates - 10X Genomics](https://www.10xgenomics.com/resources/analysis-guides/differential-gene-expression-analysis-in-scrna-seq-data-between-conditions-with-biological-replicates)
- [Harvard-Chan Bioinformatics Core tutorial on differential expression analysis with DESeq2](https://hbctraining.github.io/scRNA-seq/lessons/pseudobulk_DESeq2_scrnaseq.html)
- [OSCA chapter on DE analyses between conditions](http://bioconductor.org/books/release/OSCA.multisample/multi-sample-comparisons.html)

### Differential abundance analysis

- [Cell type composition analysis: comparison of statistical methods - Simmons (2022)](https://doi.org/10.1101/2022.02.04.479123)
- [OSCA chapter on changes in cluster abundance](http://bioconductor.org/books/release/OSCA.multisample/differential-abundance.html)
- [scCODA is a Bayesian model for compositional single-cell data analysis - Buttner _et al._ (2021)](https://doi.org/10.1038/s41467-021-27150-6)
- [scDC: single cell differential composition analysis - Cao _et al._ (2019)](https://doi.org/10.1186/s12859-019-3211-9)
